README.md

```diff
@@ -235,6 +236,17 @@ $ kubectl get events
   9m45s       Warning   UpdateFailed        nexus/nexus3                   Failed to update to 3.26.1. Human intervention may be required
 # (output omitted)
 ```
+## Custom Configuration
+
+Starting on version 0.6.0, now the operator mounts a [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) with
+the contents of the [`nexus.properties`](https://help.sonatype.com/repomanager3/installation/configuring-the-runtime-environment) file
+in the path `$NEXUS_DATA/etc/nexus.properties`. The `ConfigMap` has the same name as your Nexus CR.
+
+The Nexus Operator mount this file with empty contents, but you can update it anytime with the desired configuration. 
```

Comment:
Also, maybe instead of mounting it empty and making the user populate it every time we can instead receive this CM as parameter via a spec field, wdyt?

I'm assuming it will be common for new deployments of the Nexus to use the same configuration as the previous, if we reset it every time this will become a pretty bad toil for our users.

Reply:
> Also, maybe instead of mounting it empty and making the user populate it every time we can instead receive this CM as parameter via a spec field, wdyt?

Yeah, I thought about this possibility but feels awkward to have the configuration in the CR, basically replicating the same information twice can be confused. But can come in handy, yes. wdyt @ibotty?

> I'm assuming it will be common for new deployments of the Nexus to use the same configuration as the previous, if we reset it every time this will become a pretty bad toil for our users.

Not sure if I followed. The ConfigMap will have to change once **if** they need any custom configuration, otherwise, it won't get redeployed nor the pod will rollout. Once we get HA, this problem will be solved. :p

Reply:
If you redeploy the Nexus service, the configMap remains the same. They won't need to update anything. The operator will load the current deployed CM and will leave it there, binding it to the pod. So in practice, a user only has to add their configuration once.

Reply:
Yes I was hallucinating after a long day, the map won't be changed by the operator

Reply:
@LCaparelli I think it would be nice to have something like:

```yaml
spec:
    properties:
         nexus.my.property: myvalue
```

So the source of truth would be the Nexus CR instead of the CM. The CM then would be fully managed by the Operator. If a user would like to change anything, they would change the CR instead. So we keep the state and facilitate migrations.

@ibotty wdyt? Since you brought this use case, would be valuable to have your inputs!

Reply:
My suggestion was to receive a ref to the ConfigMap (something like `spec.propertiesMap` or some better name along that line), but defining a string key-value dict inline like you're suggesting also works quite nicely. Note these alternatives don't exclude each other: we can either have the inline definition of these properties or receive a reference to a ConfigMap (should help with re-using maps for multiple deployments), though we should validate we don't do both. These are separate features though and can be implemented separately, just wanted to check if this is a path we want to follow.

Another point is that I think UX will be better if each key of this map (the one we receive or the inlined one) should be a property, not a single key pointing to all config. Wdyt?

The caveat I see is that we'd need to parse it all into a single file to mount it properly, but shouldn't be too much work or too complex.

Reply:
> Another point is that I think UX will be better if each key of this map (the one we receive or the inlined one) should be a property, not a single key pointing to all config. Wdyt?

That's the idea:

```yaml
spec:
    properties:
         nexus.my.property1: myvalue1
         nexus.my.property2: myvalue2
         nexus.my.property3: myvalue3
```

Will then be mounted as:

```properties
nexus.my.property1=myvalue1
nexus.my.property2=myvalue2
nexus.my.property3=myvalue3
```

Strimzi is already doing this, not sure if they mount the CM, but they receive the properties like this in their CR. I think it's pretty cool.

Reply:
+1! Saw that on Strimzi too hehe

Reply:
I'll work on it and update the PR.